src/argilla/client/feedback/training/base.py

```diff
@@ -71,10 +71,20 @@ def __init__(
             gpu_id: the GPU ID to use when training a SpaCy model. Defaults to -1, which means that the CPU
                 will be used by default. GPU IDs start in 0, which stands for the default GPU in the system,
                 if available.
+            filter_by: A dict with key the field to filter by, and values the filters to apply. Currently only
+                defined for `response_status` filters. Can be one of: draft, pending, submitted, and discarded.
+                Defaults to `None` (no filter is applied).
             framework_kwargs: arguments for the framework's trainer. A special key (model_card_kwargs) is reserved
                 for the arguments that can be passed to the model card.
             **load_kwargs: arguments for the rg.load() function.
         """
+        if filter_by:
+            # Use the id attribute of RemoteFeedbackDataset to determine the type of dataset.
+            # There are errors due to circular imports when trying to import RemoteFeedbackDataset.
+            if not getattr(dataset, "id", None):
```

Comment:
I think a `isinstance` check for `RemoteFeedbackDataset` might be better.

Reply:
I also think that's the best option, but I kept errors regarding circular options. Maybe we can review this problem tomorrow?

Reply:
There isn't really a need for that check after all, it's removed now.